src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/interrupts/CheckForInterruptsState.java

```diff
@@ -132,6 +132,8 @@ public void activate() {
     }
 
     public void deactivate() {
+        /* avoid race condition caused by interference from interruptChecks thread */
+        shouldTrigger = true;
```

Comment:
wait, why do we set shouldTrigger to true here, and then to false?

Reply:
It may be unnecessary... I just wanted to make sure that if interruptChecks fires, it would see that shouldTrigger is already true and not potentially change shouldTrigger to true in the other thread.